dynamicio/io/resource.py

```diff
@@ -0,0 +1,97 @@
+from abc import abstractmethod
+from copy import deepcopy
+from pathlib import Path
+from typing import Callable, List, Optional, Type
+
+import pandas as pd
+from pandera import SchemaModel
+from pydantic import BaseModel
+from uhura import Readable, Writable
+
+from dynamicio.inject import InjectionError, inject
+from dynamicio.io.serde import BaseSerde, PickleSerde
+
+
+def create_schema_validator(schema) -> Callable[[pd.DataFrame], pd.DataFrame]:
+    def validate_schema(df: pd.DataFrame):
+        return schema.validate(df)
+
+    return validate_schema
+
+
+class BaseResource(BaseModel, Readable[pd.DataFrame], Writable[pd.DataFrame]):  # Cacheable kaputt, ABC kaputt
```

Comment:
remove --> # Cacheable kaputt, ABC kaputt

Reply:
Adding ABC to inheritance.